TACC-Cloud/hazmapper
# task/DES-1828 Streetview

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [DES-1839](https://jira.tacc.utexas.edu/browse/DES-1839)
* [DES-1828](https://jira.tacc.utexas.edu/browse/DES-1828)
* [DES-1893](https://jira.tacc.utexas.edu/browse/DES-1893)


## Summary of Changes: ##
* [x] Separate panel for Streetview
  * [x] Tabs for Accounts, Upload Log, and Uploaded assets (streamed from Streetview service)
* [x] Upload Log
  * [X] Separate status indicators for notification types 
  * [X] Progress bar
* [x] Accounts tab for indicating authorization status
  * [x] Auth routing 
* [x] Assets
  * [x] Get mapillary sequence data from geoapi
  * [x] Notify user of pending approval
  * [x] Implement changesets for mapillary 
* [x] Streetview publishing modal
  * [x] Select service (maybe checkbox?
* [X] Display sequence features in the map
  * [x] Make each feature a marker layer 
  * [x] Open link to streetview on clicking point (with image key)
  * [x] Get thumbs from mapillary    
  * [X] Lon/lat based mapillary open
  * [X] Split mapillary view

## Testing Steps: ##
 - Use TACC-Cloud/geoapi#46 
 - Create a Mapillary account 
 - Test that uploads work for streetview images from PRJ-2996
 - Test that linking (just associating streetivew images with existing sequences) works
 - Check that uploaded/linked sequences appear in the asset list and on the map as LineStrings
 - Check that the the split views work (right click on a LineString point).

## UI Photos:
![HazMapper - Google Chrome_002](https://user-images.githubusercontent.com/9425579/113579943-99ca7400-95ea-11eb-9276-b86267cbc45b.png)
![HazMapper - Google Chrome_001](https://user-images.githubusercontent.com/9425579/113579981-a6e76300-95ea-11eb-84c8-0e0d1af22c8a.png)
![HazMapper - Google Chrome_005](https://user-images.githubusercontent.com/9425579/113580003-ae0e7100-95ea-11eb-9bd7-e90f1487a635.png)
![HazMapper - Google Chrome_008](https://user-images.githubusercontent.com/9425579/113580015-b1096180-95ea-11eb-898d-197f3d5b0172.png)




## Notes: ##
